framework/src/Volo.Abp.Ddd.Domain/Volo/Abp/Domain/Entities/Caching/AbstractKeyEntityCache.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System.Threading.Tasks;
+using Volo.Abp.Caching;
+using Volo.Abp.Domain.Entities.Events;
+using Volo.Abp.Domain.Repositories;
+using Volo.Abp.EventBus.Distributed;
+using Volo.Abp.ObjectMapping;
+
+namespace Volo.Abp.Domain.Entities.Caching
+{
+    public abstract class AbstractKeyEntityCache<TEntity, TCacheItem, TKey> :
+        IDistributedEventHandler<EntityChangedEventData<TEntity>>,
+        IEntityCache<TCacheItem, TKey>
+        where TEntity : class, IEntity
+        where TCacheItem : class
+    {
+        public TCacheItem this[TKey key] => Get(key);
+
+        public string CacheName { get; }
+
+        public IObjectMapper ObjectMapper { get; set; }
```

Comment:
Property injecting the `IObjectMapper` is not good. It doesn't cover the scenario where we want to use a [contextualized mapper](https://docs.abp.io/en/abp/latest/Object-To-Object-Mapping#iobjectmapper-interface).